crypto/crypto.c

```diff
@@ -435,8 +435,7 @@ int crypto_unregister(uint32_t driverid, int alg)
   /* Sanity checks. */
 
   if (driverid >= crypto_drivers_num || crypto_drivers == NULL ||
-      ((alg <= 0 || alg > CRYPTO_ALGORITHM_MAX) &&
-      alg != CRYPTO_ALGORITHM_MAX + 1) ||
+      (alg < 0 || alg > CRYPTO_ALGORITHM_MAX) ||
```

Comment:
but alg == CRYPTO_ALGORITHM_MAX + 1 mean to unregister all algo.

Reply:
CRYPTO_ALGORITHM_MAX should be the upper-limit by any logical means ; as by programming logic in C we index everything from [0, N( boundary.

Reply:
The new fix will provide later.